util/certs/certs.go

```diff
@@ -84,3 +84,17 @@ func DecodePrivateKeyPEM(encoded []byte) (*rsa.PrivateKey, error) {
 
 	return x509.ParsePKCS1PrivateKey(block.Bytes)
 }
+
+// DecodePKCS8PrivateKeyPEM attempts to return a decoded key or nil
+// if the encoded input does not contain a private key.
+func DecodePKCS8PrivateKeyPEM(encoded []byte) (interface{}, error) {
```

Comment:
Would it make sense to implement this as fallback logic in `DecodePrivateKeyPEM()` instead? It would save having to handle the fallback anywhere we are calling this today.

I'm also wondering if a return type of crypto.PrivateKey might provide a bit more of a descriptive return type than `interface{}` directly? Similar with other uses of `interface{}` in these changes

Reply:
Should we also add x509.ParseECPrivateKey to the list of fallback parsers just to be complete?

Reply:
This was separated also to keep the same function signature to keep go-apidiff job happy. I had initially put it as part of the DecodePrivateKeyPEM but that was breaking the API.

Reply:
and the functions under the hood return an interface, https://godoc.org/crypto/x509#ParsePKCS8PrivateKey . so we have to use interface{} . I can add the ParseECPrivateKey too, sure

Reply:
The go-apidiff job is non-blocking and is meant more for helping inform reviewers of potential issues. I believe it's safe to ignore it in this case since it should not break existing consumers of the api.